project-base/tests/FrontendApiBundle/Functional/Advert/GetAdvertsTest.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\FrontendApiBundle\Functional\Advert;
+
+use Ramsey\Uuid\Uuid;
+use Shopsys\FrameworkBundle\Model\Advert\Advert;
+use Tests\FrontendApiBundle\Test\GraphQlTestCase;
+
+class GetAdvertsTest extends GraphQlTestCase
+{
+    public function testGetArticles(): void
+    {
+        foreach ($this->getAdvertsDataProvider() as $dataSet) {
+            [$query, $expectedAdvertsData] = $dataSet;
+
+            $graphQlType = 'adverts';
+            $response = $this->getResponseContentForQuery($query);
+            $this->assertResponseContainsArrayOfDataForGraphQlType($response, $graphQlType);
+            $responseData = $this->getResponseDataForGraphQlType($response, $graphQlType);
+
+            $this->assertCount(count($expectedAdvertsData), $responseData);
+            foreach ($responseData as $advertData) {
+                $this->assertArrayHasKey('uuid', $advertData);
+                $this->assertTrue(Uuid::isValid($advertData['uuid']));
+
+                $this->assertKeysAreSameAsExpected(
+                    [
+                        'name',
+                        'type',
+                        'positionName',
+                        'code',
+                        'link',
+                    ],
+                    $advertData,
+                    array_shift($expectedAdvertsData)
+                );
+            }
+        }
+    }
+
+    /**
+     * @param array $keys
+     * @param array $actual
+     * @param array $expected
+     */
+    private function assertKeysAreSameAsExpected(array $keys, array $actual, array $expected): void
```

Comment:
The same method is in articles and article test. Does it make sense to move it into parent GraphQlTestCase?

Reply:
This will be removed and used `assertSame`